File: 백준/Bronze/22341. 사각형 면적/사각형 면적.c
#include <stdio.h>

int main() {
    int N, C;
    scanf("%d %d", &N, &C);
    int A = N, B = N;
    for (int i = 0; i < C; i++) {
        int X, Y;
        scanf("%d %d", &X, &Y);
        if (X >= A || Y >= B) {
            continue;
        }
        int horizonCut = X * B;
        int verticalCut = Y * A;
        if (horizonCut >= verticalCut) {
            A = X;
        } else {
            B = Y;
        }
    }
    printf("%d\n", A * B);
    return 0;
}

File: 백준/Bronze/22341. 사각형 면적/사각형 면적.py
import sys
input = lambda: sys.stdin.readline().rstrip()


def main():
    N, C = map(int, input().split())
    # 세로가 A, 가로가 B
    A, B = N, N
    for _ in range(C):
        X, Y = map(int, input().split())
        # 가로
        horizonCut = (0 if X >= A else X)*B
        # 세로
        verticalCut = (0 if Y >= B else Y)*A
        if horizonCut >= verticalCut:
            A = X
        else:
            B = Y
    print(A*B)


if __name__ == '__main__':
    main()